geo-types/src/lib.rs

```diff
@@ -48,14 +49,25 @@ extern crate rstar;
 #[macro_use]
 extern crate approx;
 
+#[deprecated(since = "0.7", note = "use `CoordFloat` or `CoordNum` instead")]
+pub trait CoordinateType: Num + Copy + NumCast + PartialOrd + Debug {}
+// Little bit of a hack to make to make this work
+#[allow(deprecated)]
+impl<T: Num + Copy + NumCast + PartialOrd + Debug> CoordinateType for T {}
+
 /// The type of an x or y value of a point/coordinate.
 ///
-/// Floats (`f32` and `f64`) and Integers (`u8`, `i32` etc.)
-/// implement this. Many algorithms only make sense for
-/// Float types (like area, or length calculations).
-pub trait CoordinateType: Num + Copy + NumCast + PartialOrd {}
-// Little bit of a hack to make to make this work
-impl<T: Num + Copy + NumCast + PartialOrd> CoordinateType for T {}
+/// Floats (`f32` and `f64`) and Integers (`u8`, `i32` etc.) implement this.
+///
+/// For algorithms which only make sense for floating point, like area or length calculations,
+/// see [CoordFloat](trait.CoordFloat.html).
+#[allow(deprecated)]
+pub trait CoordNum: CoordinateType + Debug {}
```

Comment:
Great idea! I was thinking recently about renaming `CoordinateType` recently to something like `CoordScalar`, but I wasn't sure how accurate that is, or if it makes sense since "scalar" is usually used in vectors. But I like `CoordNum`! Seems much simpler